### PR TITLE
[SENDMAIL] Fix Send To -> Desktop (create shortcut) behavior

### PR DESCRIPTION
**Pull request by TAN-Gaming**
## Proposed changes
- Set default working directory for shortcuts (except folders and zip files)
- Copy existing shortcut to the desktop if the source file is a shortcut
- Prevent destination file name collision

Notes: Verified on Windows XP SP3 and Windows 7 SP1.

Shortcut to EXE file (Before):
[ReactOS-sendto-before.webm](https://user-images.githubusercontent.com/86486978/204141743-0c683da0-04fc-4c5c-8aa5-6dc7bc5444ae.webm)

Shortcut to EXE file (After):
[ReactOS-sendto-after.webm](https://user-images.githubusercontent.com/86486978/204141767-8c38c888-2037-4f89-b38c-d3de0cd04874.webm)

### CODE REVIEW

File: dll/shellext/sendmail/CDeskLinkDropHandler.cpp [hunk 1]
```diff
@@ -87,13 +87,40 @@ CDeskLinkDropHandler::Drop(IDataObject *pDataObject, DWORD dwKeyState,
         if (SHGetPathFromIDListW(pidl, szSrc))
         {
             CStringW strTitle;
+            LPWSTR pszSourceExt;
```
Comment (HBelusca) [inline dll/shellext/sendmail/CDeskLinkDropHandler.cpp:90]:
```suggestion
            LPCWSTR pszSourceExt;
```

File: dll/shellext/sendmail/CDeskLinkDropHandler.cpp [hunk 2]
```diff
@@ -86,14 +86,48 @@ CDeskLinkDropHandler::Drop(IDataObject *pDataObject, DWORD dwKeyState,
...
+
+            if (PathFileExistsW(szDest))
+            {
+                CStringW strName(PathFindFileNameW(szDest));
+                PathYetAnotherMakeUniqueName(szDest, szDir, NULL, strName);
+            }
+
+            if (bIsLink)
+            {
+                hr = (CopyFileW(szSrc, szDest, TRUE) ? S_OK : E_FAIL);
```
Comment (learn-more) [inline dll/shellext/sendmail/CDeskLinkDropHandler.cpp:117]: What happens here if the link is already on the desktop, and you copy the link to the desktop again?

File: dll/shellext/sendmail/CDeskLinkDropHandler.cpp (hunk 1 above)
Comment (TAN-Gaming) [reply dll/shellext/sendmail/CDeskLinkDropHandler.cpp]: Done.

File: dll/shellext/sendmail/CDeskLinkDropHandler.cpp (hunk 2 above)
Comment (TAN-Gaming) [reply dll/shellext/sendmail/CDeskLinkDropHandler.cpp:117]: Capture2

**Review by learn-more — APPROVED** [review]

**Review by HBelusca — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]

**Pull request merged**